packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkCodecUtilities.ts

```diff
@@ -176,8 +177,14 @@ export interface ChunkDecoder {
 	/**
 	 * Read from stream, updating the offset.
 	 *
+	 * For fields that support incremental encoding, the encoded data can be retrieved via `getIncrementalFieldBatch`.
+	 *
 	 * @returns a TreeChunk made from the data from `stream`.
 	 * This chunk is allowed to reference/take ownership of content it reads from the stream.
 	 */
-	decode(decoders: readonly ChunkDecoder[], stream: StreamCursor): TreeChunk;
+	decode(
```

Comment:
I don't think you should change the ChunkDecoder abstraction here.

The existing pattern where a chunk decoder is an object that has all the context it needs to decode chunks of its given shape from the stream seems like it should still work fine. You just need a new ChunkDecoder for your new EncodedChunkShape.

if it needs extra context, it should get it from the DecoderContext which it can take in via its constructor like TreeDecoder does.

Reply:
Good suggestion, updated it.